### PR TITLE
HIVE-21706: REPL Dump with concurrent drop of external table fails with InvalidTableException.

### PR DESCRIPTION
**Pull request by sankarh**

### CODE REVIEW

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java [hunk 1]
```diff
@@ -712,6 +712,64 @@ public void testExternalTableDataPath() throws Exception {
...
+    replica.load(replicatedDbName, tupleBootstrap.dumpLocation, loadWithClause);
+
+    // Insert a row into "t1" and create another external table using data from "t1".
+    primary.run("use " + primaryDbName)
+            .run("insert into table t1 values (2)")
+            .run("create external table t2 as select * from t1");
+
+    // Inject a behavior so that getTable returns null for table "t1". This ensures the table is
+    // skipped for data files listing.
+    BehaviourInjection<Table, Table> ptnedTableNuller = new BehaviourInjection<Table, Table>() {
```
Comment (maheshk114) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java]: ptnedTableNuller is not proper.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java [hunk 2]
```diff
@@ -133,7 +134,17 @@ void dataLocationDump(Table table)
           PathBuilder.fullyQualifiedHDFSUri(table.getDataLocation(), FileSystem.get(hiveConf));
       write(lineFor(table.getTableName(), fullyQualifiedDataLocation, hiveConf));
       if (table.isPartitioned()) {
-        List<Partition> partitions = Hive.get(hiveConf).getPartitions(table);
+        List<Partition> partitions;
+        try {
+          partitions = Hive.get(hiveConf).getPartitions(table);
+        } catch (HiveException e) {
+          if (e.getCause() instanceof NoSuchObjectException) {
+            // If table is dropped when dump in progress, just skip partitions data location dump
```
Comment (maheshk114) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java]: can add a info log

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 3]
```diff
@@ -256,19 +256,25 @@ private Long incrementalDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive
...
+            // Dump the table to be bootstrapped if required.
+            if (shouldBootstrapDumpTable(table)) {
+              HiveWrapper.Tuple<Table> tableTuple = new HiveWrapper(hiveDb, dbName).table(table);
+              dumpTable(dbName, tableName, validTxnList, dbRoot, bootDumpBeginReplId, hiveDb,
+                      tableTuple);
+            }
+          } catch (InvalidTableException te) {
+            // Repl dump shouldn't fail if the table is dropped/renamed while dumping it.
+            // Just log a debug message and skip it.
+            LOG.debug(te.getMessage());
```
Comment (maheshk114) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: can be changed to info with table name

Comment (sankarh) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: In bootstrap dump is is debug log... also, the exception message have the table name. do you think, we need to change that as well to info?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java (hunk 2 above)
Comment (sankarh) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java]: Can be debug log?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java (hunk 1 above)
Comment (sankarh) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java]: done

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java (hunk 2 above)
Comment (sankarh) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplExternalTables.java]: done

**Pull request closed without merging**